cms/static/js/views/license-view.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Provide license display view for license models.
+ * The license display view is used to preview the selected license in
+ * forms for new or existing courses and videos. Additionally, the display
+ * view is used to preview the license underneath videos and inside the
+ * course outline. Such that the author knows under which license the
+ * course material will be published.
+ */
+define(["js/views/baseview", "underscore", "gettext", "js/models/license"],
+    function(BaseView, _, gettext, LicenseModel) {
+
+        var LicenseView = BaseView.extend({
+            initialize: function(options) {
+                if (!this.model) {
+                    this.model = new LicenseModel();
+                }
+                else if (!(this.model instanceof LicenseModel)) {
+                    if (this.model.get('license') instanceof LicenseModel) {
+                        this.model = this.model.get('license');
+                    } else {
+                        this.model = new LicenseModel({kind: this.model.get('license')});
+                    }
+                }
+
+                // Rerender on model change
+                this.listenTo(this.model, 'change', this.render);
+                this.render();
+            },
+
+            render: function() {
+                this.$el.html(this.renderLicense());
```

Comment:
Why not just put the contents of the `renderLicense` function into this `render` function, and delete the `renderLicense` function?